rq/worker.py

```diff
@@ -625,15 +625,24 @@ def monitor_work_horse(self, job):
         either executes successfully or the status of the job is set to
         failed
         """
+
+        ret_val = None
         while True:
             try:
+                job.started_at = job.started_at or utcnow()
                 with UnixSignalDeathPenalty(self.job_monitoring_interval, HorseMonitorTimeoutException):
                     retpid, ret_val = os.waitpid(self._horse_pid, 0)
                 break
             except HorseMonitorTimeoutException:
                 # Horse has not exited yet and is still running.
                 # Send a heartbeat to keep the worker alive.
                 self.heartbeat(self.job_monitoring_interval + 5)
+
+                # Kill the job from this side if something is really wrong (interpreter lock/etc).
+                if (utcnow() - job.started_at).total_seconds() > (job.timeout * 1.1):
```

Comment:
Instead of adding a 10% time multiplier, I think it's safe to add just 1 or 2 seconds to the timeout. The default job timeout is 180 so having a 10% tolerance would be 18 seconds, which is way too long.

Reply:
Sounds good. I'll retest with 1 or 2 seconds.